spec/finder_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'array_database'
+require 'finder'
+require 'user_interface'
+require 'validator'
+
+RSpec.describe Finder do
+  describe '#run' do
+    let(:database) { ArrayDatabase.new }
+    let(:output) { StringIO.new }
+    let(:validator) { Validator.new }
+
+    it 'askes user for a search_term' do
+      input = StringIO.new("term\nn\n")
+      user_interface = UserInterface.new(input, output, validator)
+      finder = described_class.new(user_interface, database)
+
+      finder.run
+
+      expect(output.string).to include(UserInterface::SEARCH_MESSAGE)
+    end
+
+    it 'prints message if no contacts are found' do
+      input = StringIO.new("term\nn\n")
+      user_interface = UserInterface.new(input, output, validator)
+      finder = described_class.new(user_interface, database)
+
+      database.create(test_contact)
+
+      finder.run
+
+      expect(output.string).to include(UserInterface::NO_CONTACTS_MESSAGE)
+    end
+
+    it 'does not print a no contacts message if contacts are found' do
+      input = StringIO.new("damon\nn\n")
+      user_interface = UserInterface.new(input, output, validator)
+      finder = described_class.new(user_interface, database)
+
+      database.create(test_contact)
+
+      finder.run
+
+      expect(output.string).not_to include(UserInterface::NO_CONTACTS_MESSAGE)
+    end
+
+    it 'prints the contacts that are found' do
+      input = StringIO.new("damon\nn\n")
+      user_interface = UserInterface.new(input, output, validator)
+      finder = described_class.new(user_interface, database)
+
+      database.create(test_contact)
+
+      finder.run
+
+      expect(output.string).to include(%(
+Name:    Matt Damon
```

Comment:
You tested this structure in the user interface, you don't need to assert the whole structure, you can just assert the line that contains the search term:

```ruby
expect(output.string).to include('Name:    Matt Damon')
```

always think how many places would you have to do changes to if something changed somewhere else.